relay-event-normalization/src/normalize/span/description/sql/parser.rs

```diff
@@ -344,3 +381,62 @@ impl Dialect for DialectWithParameters {
         self.0.parse_statement(parser)
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use sqlparser::parser::Parser;
+
+    use super::*;
+
+    #[test]
+    fn depth() {
+        let query = "SELECT (a + b + (c1 + (c2 + c3)) + c)";
+        let mut parsed = Parser::parse_sql(&GenericDialect {}, query).unwrap();
+        let mut visitor = NormalizeVisitor::new();
+        parsed.visit(&mut visitor);
+        assert_eq!(visitor.max_depth, 8);
+    }
+
+    #[test]
+    fn depth_nested_from() {
+        let query = "SELECT * FROM (SELECT 1)";
+        let mut parsed = Parser::parse_sql(&GenericDialect {}, query).unwrap();
+        let mut visitor = NormalizeVisitor::new();
+        parsed.visit(&mut visitor);
+        assert_eq!(visitor.max_depth, 1);
+    }
+
+    #[test]
+    fn depth_nested_from_2() {
+        let query = "SELECT * FROM (SELECT * FROM (SELECT 1))";
+        let mut parsed = Parser::parse_sql(&GenericDialect {}, query).unwrap();
+        let mut visitor = NormalizeVisitor::new();
+        parsed.visit(&mut visitor);
+        assert_eq!(visitor.max_depth, 2);
+    }
+
+    #[test]
+    fn depth_nested_select() {
+        let query = "SELECT (SELECT 1) AS one FROM a";
+        let mut parsed = Parser::parse_sql(&GenericDialect {}, query).unwrap();
+        let mut visitor = NormalizeVisitor::new();
+        parsed.visit(&mut visitor);
+        assert_eq!(visitor.max_depth, 2);
+    }
+
+    #[test]
+    fn depth_nested_select_2() {
+        let query = "SELECT (SELECT (SELECT 1)) AS one FROM a";
+        let mut parsed = Parser::parse_sql(&GenericDialect {}, query).unwrap();
+        let mut visitor = NormalizeVisitor::new();
+        parsed.visit(&mut visitor);
+        assert_eq!(visitor.max_depth, 3);
+    }
+
+    #[test]
+    fn parse_deep_expression() {
+        let query = "SELECT 1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1";
```

Comment:
From this tests, it seems to me that `MAX_DEPTH` is the number of element in the query we parse? like `SELECT` is one and than each `1` and `+` and  two more and so forth ?

Reply:
It's supposed to reflect the depth of the abstract syntax tree parsed by `sqlparser`. In this case, the arithmetic expression gets parsed as

```
   +
 /   \
1     +
    /   \
   1     etc.
```